plugins/inputs/ceph/ceph.go

```diff
@@ -354,37 +354,56 @@ func (c *Ceph) execute(command string) (string, error) {
 
 // CephStatus is used to unmarshal "ceph -s" output
 type CephStatus struct {
+	FSMap struct {
+		NumUp        float64 `json:"up"`
+		NumIn        float64 `json:"in"`
+		NumMax       float64 `json:"max"`
+		NumUpStandby float64 `json:"up:standby"`
+	} `json:"fsmap"`
 	Health struct {
 		Status        string `json:"status"`
-		OverallStatus string `json:"overall_status"`
+		OverallStatus string `json:"overall_status"` // Only valid for ceph version <15
 	} `json:"health"`
+	MonMap struct {
+		NumMons float64 `json:"num_mons"`
+	} `json:"monmap"`
 	OSDMap struct {
-		OSDMap struct {
```

Comment:
Did the structure of the json change? Or was this just a misconfiguration from the beginning?

Reply:
Could you try if old structure works with current ceph?

If not, it will need to be reverted and the new fields will need to be added at the upper level.

Reply:
Yes that was indeed what I was thinking as well…

Reply:
Latest changes seem to capture both situations, but I'm certain the code to do it could be optimized